app/components/common/rdsCalendar/index.tsx

```diff
@@ -23,31 +25,44 @@ const RdsCalendar = ({
   eventsList,
   defaultDate,
   currentEvent,
-  setCalendarEvent,
   updateEvent,
 }: RdsCalendarProps) => {
+  const [date, setDate] = React.useState(dayjs(currentEvent?.start).toDate());
   const RbcCalendar = withDragAndDrop(Calendar);
   const onEventDrop: withDragAndDropProps['onEventDrop'] = (ev: UpdateEvent) =>
     updateEvent({ ...ev.event, start: dayjs(ev.start).toDate(), end: dayjs(ev.end).toDate() });
   const onEventResize: withDragAndDropProps['onEventResize'] = (ev: UpdateEvent) =>
     updateEvent({ ...ev.event, start: dayjs(ev.start).toDate(), end: dayjs(ev.end).toDate() });
+  const navigate = useNavigate();
+  const setView = useStore((state) => state.setView);
+
+  const handleNavigate = (localDate: any) => {
+    setDate(dayjs(localDate).toDate());
+  };
 
   return (
     <RbcCalendar
       localizer={localizer}
       events={eventsList}
       defaultDate={defaultDate}
-      defaultView={view ?? 'week'}
+      defaultView={view}
       style={{ height: height ?? '100vh' }}
-      onSelectEvent={(event) => setCalendarEvent((e) => ({ ...e, event, show: true, new: false }))}
+      onSelectEvent={(event: CalEvent) => {
+        navigate(`/event/${event.id}`);
+      }}
+      date={date}
       onEventDrop={onEventDrop}
       onEventResize={onEventResize}
       selectable={true}
       scrollToTime={dayjs(currentEvent?.start).toDate()}
-      onSelectSlot={({ start }) => {
-        const event: CalEvent = { title: '', start, end: dayjs(start).add(1, 'hour').toDate() };
-        setCalendarEvent((e) => ({ ...e, event, show: true, new: true }));
+      onSelectSlot={(ev: CalEvent) => {
+        navigate('/event/new', {
+          state: { start: dayjs(ev.start).toDate(), end: dayjs(ev.end).toDate() },
+        });
       }}
+      onView={(v: View) => setView(v)}
+      // onRangeChange={(e) => console.log(e)}
```

Comment:
What is the purpose of onRangeChange here?
Could we remove this and add it as an issue to look into it later?